src/Microsoft.DotNet.Interactive.HttpRequestParser/HttpRequestParser.cs

```diff
@@ -0,0 +1,418 @@
+                if (CurrentToken.Kind is HttpTokenKind.Whitespace)
+                {
+                    ConsumeCurrentTokenInto(node);
+                }
+                else if (CurrentToken.Kind is HttpTokenKind.NewLine)
+                {
+                    ConsumeCurrentTokenInto(node);                   
+                }
+                else if (CurrentToken is { Kind: HttpTokenKind.Punctuation } and { Text: "#" })
+                {                                   
+                    node.Add(ParseComment());
+                } 
+                else if (CurrentToken is { Kind: HttpTokenKind.Punctuation } and { Text: "/" } && 
+                    NextToken is { Kind: HttpTokenKind.Punctuation } and { Text: "/" })
+                {
+                    node.Add(ParseComment());
+                }
+                else
+                {
+                    break;
+                }
+            }
+
+            return node;
+        }
+
+        private T ParseTrailingTrivia<T>(T node, bool stopAfterNewLine = false) where T : HttpSyntaxNode
+        {
+            while (MoreTokens())
+            {
+                if (stopAfterNewLine && CurrentToken.Kind is HttpTokenKind.NewLine)
+                {
+                    ConsumeCurrentTokenInto(node);
+                    break;
+                }
+
+                if (CurrentToken.Kind is not (HttpTokenKind.Whitespace or HttpTokenKind.NewLine))
+                {
+                    break;
+                }
+
+                ConsumeCurrentTokenInto(node);
+            }
+
+            return node;
+        }
+
+        private HttpRequestNode ParseRequest()
+        {
+            var methodNode = ParseMethod();
+            var urlNode = ParseUrl();
+            var versionNode = ParseVersion();
+            var headersNode = ParseHeaders();
+            var bodySeparatorNode = ParseBodySeparator();
+            var bodyNode = ParseBody();
+
+            return new HttpRequestNode(
+                _sourceText,
+                _syntaxTree,
+                methodNode,
+                urlNode,
+                versionNode,
+                headersNode,
+                bodySeparatorNode,
+                bodyNode);
+        }
+
+        private HttpMethodNode? ParseMethod()
+        {
+            var node = new HttpMethodNode(_sourceText, _syntaxTree);
+
+            ParseLeadingTrivia(node);
+
+            if (MoreTokens() && CurrentToken.Kind is HttpTokenKind.Word)
+            {
+                if (CurrentToken.Kind is HttpTokenKind.Word && CurrentToken.Text.ToLower() is ("get" or "post" or "patch" or "put" or "delete" or "head" or "options" or "trace"))
+                {                 
+                    ConsumeCurrentTokenInto(node);
+                }
+                else if (CurrentToken.Kind is HttpTokenKind.Word && CurrentToken.Text.ToLower() is not ("get" or "post" or "patch" or "put" or "delete" or "head" or "options" or "trace" or "https"))
```

Comment:
We can simplify these `if` statements to reduce repetition.

Reply:
I think we have a pattern where we repeatedly check for certain special states (for example a common start state we check for is `MoreTokens() && CurrentToken.Kind is not (TokenKind.WhileSpace or TokenKind.NewLine)`.

As we go along, it may be a good idea to pull out helpers for repeated patterns that we check for.